### PR TITLE
ADST-422

### PR DESCRIPTION
**Pull request by aman-alfresco**
- added auth0 authentication

### CODE REVIEW

**Review by PiplaniVinay — APPROVED** [review]

File: auth/src/main/java/com/alfresco/auth/AuthInterceptor.kt [hunk 1]
```diff
@@ -224,10 +225,37 @@ class AuthInterceptor(
         }
     }
 
+    private inner class OIDCProvider (val accessToken : String) : Provider{
+        override fun intercept(chain: Interceptor.Chain): Response {
+            val response = chain.proceed(AuthType.OIDC, accessToken)
+
+            // When unauthorized try to refresh
```
Comment (gl-lovekesh) [inline auth/src/main/java/com/alfresco/auth/AuthInterceptor.kt:232]: remove commented code

File: auth/src/main/java/com/alfresco/auth/DiscoveryService.kt [hunk 2]
```diff
@@ -62,6 +79,31 @@ class DiscoveryService(
...
+                    .url(URL(uri))
+                    .get()
+                    .build()
+                val response = client.newCall(request).execute()
+
+                if (response.code != 200) return@withContext false
+
+                val body = response.body?.string() ?: ""
+                val data = AppConfigDetails.jsonDeserialize(body)
+                return@withContext data?.oauth2?.audience?.isNotBlank() == true
```
Comment (gl-lovekesh) [inline auth/src/main/java/com/alfresco/auth/DiscoveryService.kt:99]:
```suggestion
                return @withContext data?.oauth2?.audience?.isNotBlank() == true
```

File: auth/src/main/java/com/alfresco/auth/AuthInterceptor.kt (hunk 1 above)
Comment (aman-alfresco) [reply auth/src/main/java/com/alfresco/auth/AuthInterceptor.kt:232]: ok

File: auth/src/main/java/com/alfresco/auth/DiscoveryService.kt (hunk 2 above)
Comment (aman-alfresco) [reply auth/src/main/java/com/alfresco/auth/DiscoveryService.kt:99]: What are the changes you suggested here?

File: auth/src/main/java/com/alfresco/auth/pkce/PkceAuthService.kt [hunk 3]
```diff
@@ -82,27 +123,100 @@ internal class PkceAuthService(context: Context, authState: AuthState?, authConf
...
+        }
+    }
+
+    private suspend fun webAuthAsync(
+        oauth2: OAuth2Data,
+        activity: Activity
+    ): Credentials? {
+        return try {
+            val host = URL(oauth2.host).host
+            Log.d("Test OIDC -1 :: ", oauth2.clientId)
```
Comment (gl-lovekesh) [inline auth/src/main/java/com/alfresco/auth/pkce/PkceAuthService.kt:175]: remove these test data.

**Review by gl-lovekesh — CHANGES_REQUESTED** [review]
please remove the unwanted code

File: auth/src/main/java/com/alfresco/auth/pkce/PkceAuthService.kt (hunk 3 above)
Comment (aman-alfresco) [reply auth/src/main/java/com/alfresco/auth/pkce/PkceAuthService.kt]: Done

**Review by gl-lovekesh — APPROVED** [review]
LGTM

**Pull request merged**